### PR TITLE
refactor: implement MVC pattern

### PR DESCRIPTION
**Pull request by Jorrre**
Reorganises the code into using the MVC pattern, mostly for the actual gameplay.

### CODE REVIEW

**Comment by Jorrre** [comment]
Waiting for #27, do not merge before #27 is merged

**Review by Jorrre — COMMENTED** [review]
Waiting for #27

**Review by matsfinsas — APPROVED** [review]

**Pull request merged**
